### PR TITLE
Add tests for resiliency feature integration

### PR DESCRIPTION
**Pull request by maanug-nv**
# What does this PR do ?

Add back tests that were originally in #11117

**Collection**: [llm]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by ShriyaPalsamudram — DISMISSED** [review]

File: nemo/lightning/megatron_parallel.py [hunk 1]
```diff
@@ -589,7 +589,7 @@ def init_ddp(self):
                     self.ddp_config,
                     module,
                     data_parallel_group=parallel_state.get_data_parallel_group(with_context_parallel=True),
-                    expert_data_parallel_group=parallel_state.get_data_modulo_expert_parallel_group(),
+                    expert_data_parallel_group=parallel_state.get_expert_data_parallel_group(),
```
Comment (maanug-nv) [inline nemo/lightning/megatron_parallel.py:592]: I don't think MCore in CI container needs this change

Comment (maanug-nv) [reply nemo/lightning/megatron_parallel.py:592]: @ShriyaPalsamudram

Comment (ShriyaPalsamudram) [reply nemo/lightning/megatron_parallel.py:592]: Given that the [deprecation warning is already out](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/core/parallel_state.py#L1712), isn't it good to change it now before it ends up getting deprecated and then throwing an error?

Comment (maanug-nv) [reply nemo/lightning/megatron_parallel.py:592]: the function doesn't exist in the current MCore commit we have in the CI container. We are using `c1728c12f1f1cdbb786e52f1ffe512295d76bef3`. `get_expert_data_parallel_group()` was introduced in `7f22e210cddc3215adda25d9e16ea512dc32458c`.

Comment (ShriyaPalsamudram) [reply nemo/lightning/megatron_parallel.py]: Oh I see, thanks for pointing this out. Reverted in the latest commit.

**Review by ShriyaPalsamudram — DISMISSED** [review]

**Review by hemildesai — APPROVED** [review]

**Review by pablo-garay — APPROVED** [review]

**Pull request merged**